loopy/codegen/__init__.py

```diff
@@ -550,4 +550,23 @@ def generate_body(kernel):
 
 # }}}
 
-# vim: foldmethod=marker
+from loopy.target.c import CASTIdentityMapper
+class FunctionDeclExtractor(CASTIdentityMapper):
+    def __init__(self):
+        self.decls = []
+
+    def map_function_declaration(self, node):
+        self.decls.append(node)
+        return super(self.__class__, self).map_function_declaration(node)
+
+def generate_header(kernel):
```

Comment:
This should be added to the documentation. Also, in a way, this is misplaced here--since C-family targets are the only ones that have a notion of headers. The fact that you have to import something from the `CTarget` here is a giveaway of that mismatch. I'd suggest moving this to the C target source file. It can still be exported to the loopy toplevel namespace (since C-family targets are a big part of what loopy deals with), but there should be adequate clarification (and perhaps a check) that this doesn't make much sense (and shouldn't be called) for non-C targets.